libraries/botframework-schema/src/teams/index.ts

```diff
@@ -1820,73 +1820,96 @@ export interface MeetingEndEventDetails extends MeetingEventDetails {
 
 /**
  * @interface
- * Specifies details of a Teams meeting send notification payload.
+ * Specifies base details of a Teams meeting send notification payload.
  */
-export interface TeamsMeetingNotification {
+export interface MeetingNotificationBase<T> {
     /**
      * @member {string} [type] The type of meeting notification.
      */
-    type: 'targetedMeetingNotification' | string;
+    type: string;
+
+    /**
+     * @template T
+     * @member {T} value The specific details of the meeting notification.
+     */
+    value: T;
+}
+
+/**
+ * @interface
+ * Specifies details of a targeted Teams meeting send notification payload.
+ */
+export interface TargetedMeetingNotification extends MeetingNotificationBase<TargetedMeetingNotificationInfo> {
     /**
-     * @member {TeamsMeetingNotificationInfo} [type] The specific details of the meeting notification.
+     * @member {string} [type] The type of meeting notification.
      */
-    value: TeamsMeetingNotificationInfo;
+    type: 'targetedMeetingNotification';
+
     /**
-     * @member {TeamsMeetingNotificationChannelData} [type] The channel data of the meeting notification.
+     * @member {MeetingNotificationChannelData} [channelData] The channel data of the meeting notification.
      */
-    channelData: TeamsMeetingNotificationChannelData;
+    channelData?: MeetingNotificationChannelData;
```

Comment:
I think channelData needs to be in the base class as any meeting notification payload will have three properties. 

(1) Type
(2) T Value
(3) ChannelData 

For targeted meeting notification payload, the value is going to be TargetedMeetingNotificationValue/Info :)

Reply:
The reason I decided to put this in targeted meeting notification is because currently it's the only concrete type that requires `channelData` property. Now there's two possibilities going forward.
1. For any future concrete types, all of them have `channelData` property either as required or optional.
2. At least one of the future concrete types would not ever require `channelData` property in the payload.

Now if we decide to move `channelData` to the base class and option 2 becomes a reality, then we run into a bit of an issue. With what I have currently, we are safe guarding against this issue. Also keep in mind that if we define any future concrete type we can simply just add `channelData` property to the object without relying on inheritance. This is because the type of the notification payload isn't `MeetingNotificationBase` but instead `MeetingNotification`.

Reply:
Happy to hop on a call;

The below is the intention that I want the code to capture. 

(1) Meeting Notification payload will have the following three properties. 
- Type (required)
- T Value  (required)
- ChannelData (optional).

(2) TargetedMeetingNotification is a part of MeetingNotification family whose type is 'TargetedMeetingNotificaiton'. 
- The unique part of TargetedMeetingNotification is that the Value always should contain `Recipients` and `Surfaces`. 

(3) Let's say we introduce another member of Meeting Notification payload whose type is 'Test'. 
- `Test` meeting notification payload will have a different structure for its `Value` (ex: it will have int and string)
- But it still has an ability to send user attributes as it is a part of Meeting Notification payload (therefore, it should have channelData as an optional parameter)

If your current implementation does allow this scenario, I am all good :)  

Coming from my OOP oriented programming language perspective, I cannot quite see how the current implementation of model classes satisfies this intention unless each child class from MeetingNotificationBase specifies ChannelData.

Reply:
I think what I have is very close to what you are describing. (1), (2) are already satisfied. The new interface `Test` will be part of the `MeetingNotification` family with `Value` being whatever type we want it to be as feature designers. This interface will also have `channelData` property explicitly defined instead of it being inherited from the base interface. This will then satisfy (3).

I think what I have is very close to what you would like. Let's hop on call soon to clarify.

Reply:
Hey @singhk97 , happy to hop on a call - will set up something :) 
I know the current design CAN satisfy the behaviors. But is it the right way of doing so when we know that any child classes of MeetingNotificationBase need to have ChannelData as an optional property? Does the current design reflect the intention of the API correctly?

Reply:
Will move the `channelData` to base class in the next iteration based on offline discussion.